components/canvas/drawingarea.tsx

```diff
@@ -3,94 +3,73 @@ import { useEffect, useRef, useState } from "react";
 import Button from "../button/Button";
 import PageLink from "../pagelink/Pagelink";
 
-let x = 1.0;
-let y = 1.0;
+function getFromLocalStorage<T>(key: string, defaultValue: T) {
+  return localStorage.getItem(key) || defaultValue;
+}
 
 function Drawing() {
-  const canvasRefFabric = useRef(null);
-  const canvasRefDesign = useRef(null);
-  const [patternImage, setPatternImage] = useState("");
-  const [chosenPattern, setChosenPattern] = useState("");
+  const [zoom, setZoom] = useState<number>(1.0);
+  const canvasRefDesign = useRef<HTMLCanvasElement>(null);
+  const uploadedImageSrc = getFromLocalStorage<string>(
+    "Image",
+    "/pusteblume.jpg"
+  );
+  const selectedPatternSrc = getFromLocalStorage<string>("Design", "");
+  useEffect(() => {
+    const context = canvasRefDesign.current.getContext("2d");
+    context.clearRect(0, 0, 302, 152);
+    const uploadedImage = new Image();
+    uploadedImage.src = uploadedImageSrc;
+    const selectedPattern = new Image();
+    selectedPattern.src = selectedPatternSrc;
+    function drawSelectedPattern() {
+      context.drawImage(selectedPattern, 0, 0, 302, 152);
+    }
+    function drawUploadedImage() {
+      context.drawImage(
+        uploadedImage,
+        0,
+        0,
+        uploadedImage.width,
+        uploadedImage.height,
+        0,
+        0,
+        uploadedImage.width * zoom,
+        uploadedImage.height * zoom
+      );
+      if (selectedPattern.complete) {
+        drawSelectedPattern();
+      } else {
+        selectedPattern.onload = drawSelectedPattern;
+      }
+    }
+    if (uploadedImage.complete) {
+      drawUploadedImage();
+    } else {
+      uploadedImage.onload = drawUploadedImage;
+    }
+  }, [uploadedImageSrc, selectedPatternSrc, zoom]);
 
   function minusClick() {
-    const image = new Image();
-    image.src = chosenPattern;
-    if (image.width * (x * 0.8) > 302) {
-      x = Number(x) * 0.8;
-      y = Number(y) * 0.8;
-      createPattern(chosenPattern);
+    const uploadedImage = new Image();
+    uploadedImage.src = uploadedImageSrc;
+    if (uploadedImage.width * (zoom * 0.8) > 302) {
+      setZoom(Number(zoom) * 0.8);
```

Comment:
```suggestion
    if (uploadedImage.width * (+zoom * 0.8) > 302) {
      setZoom(+zoom * 0.8);
```
you use 'Number(zoom)' to cast 'zoom' to a Number, right? you can do it with a simple '+' ... so your 'zoom' will work in the if-statement too.